### ISSUE TITLE
Not working on iOS 18

### ISSUE THREAD

**Issue by valinet** [Tech-How/Bing-Wallpaper-for-iOS#10]
In typical Apple fashion, they broke something in the Shortcuts flow. Now, when run, the app simply displays “There was a problem running the shortcut” and a fully green wallpaper is set. Notification and everything else works as usual, though. That green wallpaper (actually red is 0, green is 132, and blue is 0 for whole image) hints to being a problem when saving the image file itself, and the image ends up all zeros in YCbCr color space, which when converted to rgb yields that green image. Can you take a look? Thanks

**Comment by valinet** [comment]
Yeah, “Set wallpaper” action is generally broken, bummer… nothing to be done, except waiting for Apple to fix it.

**Comment by Tech-How** [comment]
Interesting. What iOS version is this on? I would try swapping out the set wallpaper action for a save file or save image action to see if it also saves a green image. Then, I would try using the set wallpaper action in another more conventional shortcut. The code still returns the correct image on my end, I wonder what Apple could have done.

**Comment by valinet** [comment]
It’s on iOS 18 beta. Already did what you suggested, still green image. Basically, they broke the “Set wallpaper” action. Waiting for a fix from them, have filled an issue in the Feedback app.

**Comment by Chandnishah97** [comment]
Is there a fix for this issue? I’ve been facing the same problem since I switched to iOS18

**Comment by Tech-How** [comment]
> Is there a fix for this issue? I’ve been facing the same problem since I switched to iOS18

Not that I am aware of, unfortunately. I've just checked and confirmed that this is an issue on Apple's part.  Other users enrolled in the beta are experiencing the same problem. The best we can do is submit feedback via their portal and wait patiently for a fix.

**Comment by Chandnishah97** [comment]
Hi so in the newest apple update, I’m able to save the wallpaper and save it and even set it as a wallpaper, but the automation/shortcut doesn’t work. Is there any solution to this?

**Comment by Tech-How** [comment]
> Hi so in the newest apple update, I’m able to save the wallpaper and save it and even set it as a wallpaper, but the automation/shortcut doesn’t work. Is there any solution to this?

This is still the green image output from the iOS 18 beta, correct?

**Comment by Chandnishah97** [comment]
No there is an actual image from Bing. When I download it and save it, it’s able to download and I’m able to set it as the wallpaper. However the automation doesn’t work it shows an error. 
![Uploading IMG_2127.png…]()

**Comment by Tech-How** [comment]
> No there is an actual image from Bing. When I download it and save it, it’s able to download and I’m able to set it as the wallpaper. However the automation doesn’t work it shows an error. 
> 
> ![IMG_2127](https://github.com/user-attachments/assets/b1edff6a-2443-4a41-8585-d3ed573503d8)
> 
> 
> 
> 

Could you kindly screenshot the error so I can look into it? It's working on iOS 17 and I am not in the iOS beta program, but I can still make changes if need be. Also, don't forget about the [automation setup](https://raw.githubusercontent.com/Tech-How/Bing-Wallpaper-for-iOS/main/images/shortcut/automation-help.png) with the number 0 being passed to the shortcut as input to prevent the menu from showing.

**Comment by AntonEmelyancev** [comment]
I recently updated to iOS 18 and automation started to fail to run this shortcut (it worked perfectly on iOS 17). Short seems to be allowed to run when device is locked, however it fails with that error. Feel free to request any additional info if needed - I'll be happy to help and diagnose it.
![photo_2024-09-23_13-22-04](https://github.com/user-attachments/assets/784c0f10-8f82-4951-b35d-bb10eb187f9b)
![photo_2024-09-23_13-21-58](https://github.com/user-attachments/assets/a9e2d9fc-407d-49a2-ae39-f6b3feea69e1)

**Comment by mjdbb1** [comment]
I just updated as well and I’m seeing the same problem with the automation not running while locked.  Like the poster above I also have the privacy settings set correctly.  Any idea what’s up?

**Comment by Tech-How** [comment]
Thank you for all the reports. I'll snag a newer device and test this ASAP.

**Comment by Tech-How** [comment]
Alright, I have just tested this running iOS 18.1.1. Here's what I found:

A **lot** has changed in regards to security permissions in Shortcuts, and the way Apple handles them. Previously, when the user had selected "Always Allow", that specific action would not prompt for consent on subsequent runs. Now, we continue getting security prompts regardless if this was clicked or not. If the input to the block is slightly different, it is treated as a new security request. At least that's what it feels like to me, based on my brief tests.

I've stripped this all the way down to just a few actions to fetch the URL and set the wallpaper, removing all UI elements and customization, and it still did not work. I received several unhelpful error messages like "Element not found", and various network errors. This was while the device was locked. When unlocked, the shortcut runs normally. I suspect that Apple has heavily restricted what shortcuts are able to do in the background. I am not able to properly understand where these error messages are coming from without being able to control and debug the Shortcuts app itself, which is not really possible. Even if it was, this is an Apple app. They control its development.

Unfortunately at the moment, I do not believe that a fix is possible, and, as in the past, we'll have to reach out to Apple or see if anything changes.

I was excited to do a revamp of this project once I upgraded my device, but I can't say I'm necessarily surprised with Apple's actions here. The potential for abuse with Shortcuts and automations is very high, and Apple is aiming for a 0% tolerance.

Still, I believe that if it's your device, you're the one responsible for securing it, and I don't think Apple's changes were necessary. You should have the flexibility to do what you wish with your electronics.

If this gets changed in the future, I may consider a rewrite to be compatible with all of these new security checks. Do keep in mind that many features I consider implementing actually get scrapped if I think the UX flow will be too noisy, and there are a lot of barriers as a result of these checks. I'm also a persistent individual, though. We'll have to see what happens from here, but I'm exhausted for resources at the moment.

For now, devices running iOS 15, 16, and (17 I think?) can still use the shortcut normally.

**Comment by mjdbb1** [comment]
I’m not sure what changed since my last posting in this thread, but it’s now been working just fine, even while locked. I’m not sure what could have happened after my last post besides restarting the phone, I didn’t change or update anything. But now it’s on he latest iOS and it’s fine. Curious that you’re seeing issues now. I really have no idea what’s going on but can anyone chime in if it’s working ok for them on the latest iOS?

**Comment by Tech-How** [comment]
> I’m not sure what changed since my last posting in this thread, but it’s now been working just fine, even while locked. I’m not sure what could have happened after my last post besides restarting the phone, I didn’t change or update anything. But now it’s on he latest iOS and it’s fine. Curious that you’re seeing issues now. I really have no idea what’s going on but can anyone chime in if it’s working ok for them on the latest iOS?

What's your current iOS version?

**Comment by mjdbb1** [comment]
18.4

**Comment by Tech-How** [comment]
Will re-test soon, anyone's open to comment.

**Comment by BrandonN5654** [comment]
Hi sorry for making a separate issue I thought they were different. For me, it does not say that it’s a privacy issue, but an error code just running the automation. I have set it up correctly. The only thing different from the example is that I renamed the shortcut, if that is the issue, please let me know.

**Comment by AntonEmelyancev** [comment]
This shortcut just started to work again couple of days ago. There was an update to 18.4.1 recently, but I'm quite certain it was fixed not exactly after update. I was traveling, my phone was unlocked at 5 AM and I noticed automation popup to provide permissions or something (even though I gave permissions back then when run it manually). I allowed everything and it works since then (even when phone is locked).

**Comment by BrandonN5654** [comment]
Just thought I’d comment and update. I deleted and reinstalled the shortcut. This time I did not change the name. I also have updated to 18.4.1. And it worked! Thanks for making a cool shortcut

**Comment by Tech-How** [comment]
The shortcut does call itself to perform certain actions, so renaming it may cause issues with that, yes. I should add a mention for that with an updated setup guide at some point.
I'll probably close this issue after some more testing if everything works the way it should

**Comment by CultureBMo** [comment]
I had been running this manually on iOS 18, but since upgrading to iOS 26 automated updates via the shortcut are working for me again!

After upgrading iOS I ran the shortcut manually and agreed to the prompt for permissions. Now my lock screen updates everyday at 6:30. 

I am now running iOS 26.1 on iPhone 14. 

Many thanks to @Tech-How for this shortcut.

**Comment by Tech-How** [comment]
I am still planning to do more re-testing on iOS 26 in the coming months as I am upgrading my device, but I'm glad to hear that certain functionality is still working!

**Comment by Tech-How** [comment]
The shortcut appears to be functional on the latest build of iOS 26.2. There's a lot of permission/Always Allow spam at the moment, I'm going to see what I can do to lessen that.

**Comment by Tech-How** [comment]
Just an update, I've tested a new implementation for managing shortcut settings and it is working. I'll have to re-write a majority of the shortcut to work with this new method, which should help reduce permission spam. Appreciate everyone's patience.
